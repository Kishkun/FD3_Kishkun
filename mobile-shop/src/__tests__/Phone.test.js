import {mount} from "enzyme";
import React from "react";
import ConnectedPhone from "../components/phone/Phone";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import { BrowserRouter as Router } from 'react-router-dom';

test("testing BasketCart component", () => {

    let phones = [
        {
            "id": "1",
            "name": "Apple"
        }
    ];

    const initialState = {
        phonePage: phones[0].id
    };
    const mockStore = configureMockStore();
    let store, wrapper;

    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}>
        <Router>
            <ConnectedPhone/>
        </Router>
    </Provider>);

    wrapper.props.match.params = 1;

    expect(wrapper).toMatchSnapshot();
});