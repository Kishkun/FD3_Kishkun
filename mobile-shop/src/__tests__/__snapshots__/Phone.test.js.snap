// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing BasketCart component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Phone)
        match={
          Object {
            "params": Object {
              "id": "1",
            },
          }
        }
      >
        <Phone
          addPhoneToBasket={[Function]}
          fetchPhoneById={[Function]}
          match={
            Object {
              "params": Object {
                "id": "1",
              },
            }
          }
        >
          <div
            className="view-container"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-9"
                />
                <div
                  className="col-md-3"
                />
              </div>
            </div>
          </div>
        </Phone>
      </Connect(Phone)>
    </Router>
  </BrowserRouter>
</Provider>
`;
